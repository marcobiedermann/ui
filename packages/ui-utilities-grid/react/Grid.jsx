import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';
import styles from '../grid.css';

class Grid extends PureComponent {
  static propTypes = {
    children: PropTypes.node,
  };

  static defaultProps = {
    children: null,
  };

  render() {
    const { children } = this.props;

    return <div className={styles.grid}>{children}</div>;
  }
}

export default Grid;
