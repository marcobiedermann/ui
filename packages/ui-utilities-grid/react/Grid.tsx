import React, { FC } from 'react';
import styles from '../grid.css';

const Grid: FC = (props) => {
  return <div className={styles.grid} {...props} />;
};

export default Grid;
