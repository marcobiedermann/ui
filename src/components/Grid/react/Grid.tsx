import React, { FC, HTMLAttributes } from 'react';
import classNames from 'classnames';
import styles from '../grid.css';

export interface GridProps extends HTMLAttributes<HTMLDivElement> {}

const Grid: FC<GridProps> = (props) => {
  const { className, ...otherProps } = props;

  return <div className={classNames(className, styles.grid)} {...otherProps} />;
};

export default Grid;
