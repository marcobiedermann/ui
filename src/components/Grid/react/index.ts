import Column from './Column';
import Grid from './Grid';
import Row from './Row';

export default Grid;
export { Column, Row };
