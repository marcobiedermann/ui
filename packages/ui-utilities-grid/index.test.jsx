/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import renderer from 'react-test-renderer';
import Grid from '.';

describe('Grid component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Grid />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
