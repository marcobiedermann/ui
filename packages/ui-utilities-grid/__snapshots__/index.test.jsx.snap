// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid component renders correctly 1`] = `
<div
  className="grid"
/>
`;
